internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -220,20 +220,21 @@ func (e *EnvStackConfig) StackName() string {
 	return NameForEnv(e.in.App.Name, e.in.Name)
 }
 
+type transformParameterFunc func(new, old *cloudformation.Parameter) *cloudformation.Parameter
+
 // transformParameters removes or transforms each of the current parameters and does not add any new parameters.
 // This means that parameters that exist only in the old template are left out.
-// The parameter`transform` is a function that transform a parameter, given its value in the new template and the old template.
-// If `old` is `nil`, the parameter does not exist in the old template.
-// `transform` should return `nil` if caller intends to delete the parameter.
-func (e *EnvStackConfig) transformParameters(
-	currParams []*cloudformation.Parameter,
-	oldParams []*cloudformation.Parameter,
-	transform func(new cloudformation.Parameter, old *cloudformation.Parameter) *cloudformation.Parameter) ([]*cloudformation.Parameter, error) {
+// The parameter`transformFunc` are functions that transform a parameter, given its value in the new template and the old template.
+// Each transform functions should keep the following in mind:
+// 1. It should return `nil` if the parameter should be removed.
+// 2. The transform functions are applied in a convolutional manner. 
```

Comment:
Sorry-- what does this mean?

Reply:
that the second filter is applied on the result of the first filter, and the third filter is applied on the result of the second filter, etc...
any suggestion on better wordings?

Reply:
🤔 I dunno...maybe "sequential"?